util/network-devp2p/src/discovery.rs

```diff
@@ -540,7 +550,8 @@ impl<'a> Discovery<'a> {
 		let ping_to = NodeEndpoint::from_rlp(&rlp.at(2)?)?;
 		let timestamp: u64 = rlp.val_at(3)?;
 		self.check_timestamp(timestamp)?;
-		let mut response = RlpStream::new_list(3);
+		let _enr_seq = rlp.val_at::<u64>(4).ok();
```

Comment:
What's the purpose of this? It seems we try to decode `u64`, but just ignore any results anyway. So that's most likely just a no-op?
Shouldn't that be used to figure out if the response len should be `3` or `4`?

Reply:
According to [PingPacket](https://github.com/ethereum/devp2p/blob/master/discv4.md#ping-packet-0x01) the `enr` field is optional so I don't it's possible to know in advance because both are part of discv4. I don't understand why though but I think discv5 is under progress.